src/pages/AddItem.js

```diff
@@ -1,23 +1,45 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useContext, useEffect } from 'react';
 import { withFirestore } from 'react-firestore';
 import AddItemHeader from './AddItemHeader';
-import getToken from '../lib/token';
+import './AddItem.css';
+import ItemError from './ItemError';
+import { ListContext } from '../listContext';
 
 const expectedPurchase = { soon: 7, kindOfSoon: 14, notSoon: 30 };
 
 const AddItem = ({ firestore }) => {
-  const [name, setName] = useState('');
+  const {
+    token,
+    duplicate,
+    setDuplicate,
+    checkForDuplicates,
+    shoppingList,
+    fetchList,
+    toggleShow,
+    error,
+    setError,
+  } = useContext(ListContext);
 
+  const [name, setName] = useState('');
   const [nextExpectedPurchase, setNextExpectedPurchase] = useState(0);
 
-  const initialToken = () => window.localStorage.getItem('token') || getToken();
-  const [token] = useState(initialToken);
   useEffect(() => {
-    window.localStorage.setItem('token', token);
-  }, [token]);
+    if (shoppingList.length === 0) {
+      fetchList(token);
+    }
+    if (error) {
```

Comment:
Desperation 😃. When we were finally able to prevent duplicates from being entered, the error message just stayed there. Maybe if we implement the suggested changes that won't be an issue.